testnet/stacks-node/src/syncctl.rs

```diff
@@ -76,34 +79,47 @@ impl PoxSyncWatchdogComms {
     }
 
     /// Wait for at least one inv-sync state-machine passes
-    pub fn wait_for_inv_sync_pass(&self, timeout: u64) -> bool {
+    pub fn wait_for_inv_sync_pass(&self, timeout: u64) -> Result<bool, burnchain_error> {
         let current = self.get_inv_sync_passes();
 
-        let now = get_epoch_time_secs();
         while current >= self.get_inv_sync_passes() {
-            if now + timeout < get_epoch_time_secs() {
-                debug!("PoX watchdog comms: timed out waiting for one inv sync pass");
-                return false;
-            }
-            sleep_ms(1000);
+            self.interruptable_sleep(timeout)?;
             std::sync::atomic::spin_loop_hint();
         }
-        return true;
+        return Ok(true);
     }
 
-    pub fn wait_for_download_pass(&self, timeout: u64) -> bool {
+    fn interruptable_sleep(&self, secs: u64) -> Result<(), burnchain_error> {
+        let deadline = secs + get_epoch_time_secs();
+        while get_epoch_time_secs() < deadline {
+            sleep_ms(1000);
+            if !self.should_keep_running() {
+                return Err(burnchain_error::CoordinatorClosed);
+            }
+        }
+        Ok(())
+    }
+
+    pub fn wait_for_download_pass(&self, timeout: u64) -> Result<(), burnchain_error> {
         let current = self.get_download_passes();
 
         let now = get_epoch_time_secs();
         while current >= self.get_download_passes() {
             if now + timeout < get_epoch_time_secs() {
                 debug!("PoX watchdog comms: timed out waiting for one download pass");
-                return false;
+                return Ok(());
```

Comment:
Shouldn't this code body just use `interruptable_sleep()`?

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/2547/commits/763bb53b2237e32e0112909078db1fd5ad6d4db8